extensions/emmet/src/abbreviationActions.ts

```diff
@@ -435,17 +442,32 @@ function expandAbbr(input: ExpandAbbreviationInput): string | undefined {
 
 	try {
 		// Expand the abbreviation
-		let expandedText = helper.expandAbbreviation(input.abbreviation, expandOptions);
+		let expandedText;
 
 		if (input.textToWrap) {
+			let parsedAbbr = helper.parseAbbreviation(input.abbreviation, expandOptions);
+			if (input.rangeToReplace.isSingleLine) {
+
+				// Fetch innermost element in the expanded abbreviation
+				let lastNode = parsedAbbr;
+				let tagName = parsedAbbr.name || '';
```

Comment:
we can move this after finding the innermost node i.e after the while loop. This way we can avoid setting the tagName multiple times inside the loop